test/framework/dump_resources.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	kubeadmbootstrapv1alpha3 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/yaml"
+)
+
+// DumpResources dump cluster API related resources to YAML
+func DumpResources(mgmt ManagementCluster, resourcePath string, writer io.Writer) error {
+	resources := map[string]runtime.Object{
+		"Cluster":             &clusterv1.ClusterList{},
+		"MachineDeployment":   &clusterv1.MachineDeploymentList{},
+		"MachineSet":          &clusterv1.MachineSetList{},
+		"Machine":             &clusterv1.MachineList{},
```

Comment:
There is also MachinePool now!

https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha3/machinepool_types.go

Reply:
fixed